src/Microsoft.AspNet.Authentication.Google/GoogleAuthenticationOptions.cs

```diff
@@ -28,5 +29,12 @@ public GoogleAuthenticationOptions()
         /// access_type. Set to 'offline' to request a refresh token.
         /// </summary>
         public string AccessType { get; set; }
+
+        /// <summary>
+        /// Defines whether access and refresh tokens should be stored in the
+        /// <see cref="ClaimsPrincipal"/> instance after a successful authentication.
+        /// This option is turned off by default to reduce the size of the final authentication cookie.
+        /// </summary>
+        public override bool SaveTokensAsClaims { get; set; } = false;
```

Comment:
Yup.

Just note that with this new pattern, OAuth2 subclasses cannot define custom notifications anymore, at least as easily as it was before (you'll probably need tons of casts if you try to define your own notifications class :smile:).